modules/tinymce/src/plugins/autolink/main/ts/api/Settings.ts

```diff
@@ -7,7 +7,11 @@
 
 import Editor from 'tinymce/core/api/Editor';
 
-const getAutoLinkPattern = (editor: Editor) => editor.getParam('autolink_pattern', /^(https?:\/\/|ssh:\/\/|ftp:\/\/|file:\/|www\.|(?:mailto:)?[A-Z0-9._%+\-]+@(?!.*@))(.+)$/i);
+const protocol = '[A-Za-z][A-Za-z\\d.+-]{0,14}:\\/\\/(?:[-.~*+=!&;:\'%@?^${}(),\\w]+@)?|www\\.|[-;:&=+$,.\\w]+@';
+const link = '[A-Za-z\\d-]+(?:\\.[A-Za-z\\d-]+)*(?::\\d+)?(?:\\/(?:[-+~=.,%()\\/\\w]*[-+~=%()\\/\\w])?)?(?:\\?(?:[-.~*+=!&;:\'%@?^${}(),\\/\\w]+))?(?:#(?:[-.~*+=!&;:\'%@?^${}(),\\/\\w]+))?';
+const defaultLinkPattern = () => new RegExp('^(' + protocol + ')(' + link + ')([-.~*+=!&;:\'%@?#^${}(),]*)$');
```

Comment:
Why wasn't the regex from polaris used here? As the Jira says it was supposed to be used and we definitely don't want this duplicated.

I recall you mentioning something about needing the groups to detect the protocol? We can likely add that to polaris if needed, but this should at it's core be using `Regexes.autolink()` by default.

Reply:
Alright, I'll change that up.